core/include/cubos/core/ecs/reflection.hpp

```diff
@@ -95,6 +96,17 @@ namespace cubos::core::ecs
             return std::move(*this);
         }
 
+        /// @brief Adds a category to a type.
```

Comment:
Might be a good idea to exemplify why this is useful. E.g., used to show related components next to each other on the UI.